src/ReplicatedStorage/Common/Util/StatefulIncrementor.lua

```diff
@@ -0,0 +1,191 @@
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local TweenService = game:GetService("TweenService")
+local Silo = require(ReplicatedStorage.Packages.Silo)
+
+--[=[
+	The StatefulIncrementor passively increments a value over a state, without spawning extra threads or timers.
+	An incrementation may be started, then collapsed or queried later.
+
+	@class StatefulIncrementor
+]=]
+local StatefulIncrementor = {}
+StatefulIncrementor.__index = StatefulIncrementor
+
+--[=[
+	Returns the current time.
+	May be overriden per-incrementor or globally.
+
+	@return number
+]=]
+function StatefulIncrementor.getTime()
+	return workspace:GetServerTimeNow()
+end
```

Comment:
This function is improperly named. It is private, so it should be ._getTime() - that said, I'm not sure this function is even useful, is there a good reason to just have a wrapper for workspace:GetServiceTimeNow()? I'm not entirely sure it's wrong, because it could be to make this part of it easily changeable, I'd just like some comments.

Reply:
It isn't actually technically private, it's public set, private read. There might be a better way to do this but I'm not sure, what do you think? It's intended to allow changing the timer used, e.g. switching from `:GetServerTimeNow()` to `os.clock()` if you wanted, since the timer sort of depends on the use case (similar to sleitnick's Timer).

Reply:
I'll approve it for now. I'd give a look at how slietnick handles it. He lets you pass it in as an optional to the constructor.

Reply:
This should be resolved, I've taken a probably better approach.